doc/design/bucketpath.rst

```diff
@@ -0,0 +1,157 @@
+==============================
+Design Document Bucket Path(s)
+==============================
+
+Introduction
+============
+
+- Purpose of the document / Problem description
+- Definitions and Acronyms
+
+Overview
+========
+
+Design Goals
+============
+
+Path Design
+===========
+
+Future Considerations
+=====================
+
+Sudo Code Outline
+=================
+
+.. code-block:: python
+
+    from exasol.bucketfs import PathBuilder
+
+    Path = PathBuilder(credentials)
+
+    udf_path = Path("bfsl://some/local/path/file.tar.gz")
+    http_bucket_path = Path("bfs://127.0.0.1:8888/service/bucket/some/file.tar.gz")
+    https_bucket_path = Path("bfss://127.0.0.1:8888/service/bucket/some/file.tar.gz")
+    chroot_path = Path("bfss://127.0.0.1:8888/service/bucket/some/sub/subsub/file.tar.gz?chroot=/some/sub/")
```

Comment:
not yet sure we need the chroot, a path to a directory would be enough I think.

Reply:
I am all on board with that, less is more for starters. But I think in the past we where discussing some use cases to make sure one can't traverse above a "pinned" directory.

Reply:
Yeah, we have this case for example with temporary directories, where the user should not write to the parent, because then it doesn't get cleaned. At the moment, this is done by not having a function to go to the parent of a location and probably by convention. So, we probably need it, when we allow the normal path to go to its parent.